src/librustc_middle/ty/sty.rs

```diff
@@ -2333,14 +2333,41 @@ impl<'tcx> Const<'tcx> {
         assert_eq!(self.ty, ty);
         let size = tcx.layout_of(param_env.with_reveal_all().and(ty)).ok()?.size;
         // if `ty` does not depend on generic parameters, use an empty param_env
-        self.eval(tcx, param_env).val.try_to_bits(size)
+        self.val.eval(tcx, param_env).try_to_bits(size)
     }
 
     #[inline]
     /// Tries to evaluate the constant if it is `Unevaluated`. If that doesn't succeed, return the
     /// unevaluated constant.
     pub fn eval(&self, tcx: TyCtxt<'tcx>, param_env: ParamEnv<'tcx>) -> &Const<'tcx> {
```

Comment:
Oh so on `Const` we have `try_eval_bits` and `eval` but they do not call each other, and there is neither `eval_bits` nor `try_eval`. But `try_eval` exists on `ConstKind`. But it is not used to implement `Const::try_eval_bits`, at least not directly... I feel I need to draw a call graph by hand to understand what happens.